packages/contracts/contracts/foundry/SphinxPluginTypes.sol

```diff
@@ -115,6 +115,7 @@ struct FoundryDeploymentInfo {
     bytes encodedDeployedContractSizes;
     uint fundsRequestedForSafe;
     uint safeStartingBalance;
+    uint safeFundsUsed;
```

Comment:
Before I review this PR more thoroughly, there are a couple things that I think need to be addressed

First, my perspective on this feature is that we’re writing un-audited code that directly feeds into logic that’s executed in the user’s Gnosis Safe. In that light, I think we need to be very cautious about how we approach this feature, and treat it more like on-chain logic than off-chain logic. In my opinion, writing un-audited logic that adds an additional `EXECUTE` action actually *decreases* the security profile of Sphinx. However, I think this is outweighed by the fact that it prevents a bug in our backend from messing up a user’s deployment midway when they use `fundSafe`. Outside of this specific situation, I think the security tradeoff points the other direction, so I don’t think we should include the `EXECUTE` leaf for scenario 2. Also, including scenario 2 in this feature appears to add complexity to the `addBalanceCheckAction` function. I think we should simplify this function as much as possible due to its mission critical nature

My second opinion is that I think we should include the `EXECUTE` action in the preview. I don’t think hiding this from the user instills trust in Sphinx. If we don’t include it, users may reasonably wonder what else Sphinx is executing through their Gnosis Safe without their knowledge.